processor/ic/src/lib.rs

```diff
@@ -64,22 +84,16 @@ async fn request_data(currency_pairs: String, opts: RequestOpts) -> String {
     let caller_principal = ic_cdk::caller();
 
     // derive the request id
-    let (random_bytes,): (Vec<u8>,) =
-        ic_cdk::call(Principal::management_canister(), "raw_rand", ())
-            .await
-            .unwrap();
-    let random_hex_byte: String = hex::encode(random_bytes)
-        .get(0..5)
-        .unwrap_or_default()
-        .to_string();
-    let request_id = format!("{}_{}", time().to_string(), random_hex_byte);
-
-    if !whitelist::is_whitelisted(caller_principal) {
-        panic!(
-            "canister with principal:{} is not allowed to call this method",
-            caller_principal
-        );
-    }
+    let request_id = generate_request_url().await;
+
+    check_gas().await;
+
+    // if !whitelist::is_whitelisted(caller_principal) {
```

Comment:
Do we disable the WhiteList feature?

Reply:
I had a call with @rsoury  about the whitelist approach